File: zoho_fetch_items.py
import requests
import json
from zoho_token_manager import get_headers, get_api_domain

organization_id = "892673756"
api_domain = get_api_domain()
headers = get_headers(organization_id)

# Endpoint for items (products/services)
url = f"{api_domain}/books/v3/items"

try:
    response = requests.get(url, headers=headers)
    response.raise_for_status()
    items = response.json()
    print("Successfully fetched items.")
    print(json.dumps(items, indent=2))
except requests.exceptions.RequestException as e:
    print(f"Error fetching items: {e}")
    if response is not None:
        print(f"Response status: {response.status_code}")
        print(f"Response body: {response.text}")

File: zoho_fetch_items.js
const axios = require('axios');
const { getHeaders, getApiDomain } = require('./zoho_token_manager');

async function fetchItems() {
    const organizationId = "892673756";
    const apiDomain = getApiDomain();
    const headers = await getHeaders(organizationId);

    // Endpoint for items (products/services)
    const url = `${apiDomain}/books/v3/items`;

    try {
        const response = await axios.get(url, { headers });
        const items = response.data;
        console.log("Successfully fetched items.");
        console.log(JSON.stringify(items, null, 2));
        return items;
    } catch (error) {
        console.log(`Error fetching items: ${error.message}`);
        if (error.response) {
            console.log(`Response status: ${error.response.status}`);
            console.log(`Response body: ${error.response.data}`);
        }
        throw error;
    }
}

// Execute the function if this file is run directly
if (require.main === module) {
    fetchItems();
}

module.exports = { fetchItems };